646/ Prevent opening select token dialog multiple times

lealobanov opened the pull request:
## Related Issue
<!-- If this PR addresses an issue, link it here (e.g., "Closes #123") -->

Fixes: If the user clicks the token button multiple times, the token list pops up multiple times. We need to ensure that it only pops up once.

https://github.com/onflow/FRW-Android/issues/646#issuecomment-2741971011

https://github.com/onflow/FRW-Android/issues/738

## Summary of Changes
<!-- Provide a concise description of the changes made in this PR. 
What functionality was added, updated, or fixed? -->

## Need Regression Testing
<!-- Indicate whether this PR requires regression testing and why. -->
- [ ] Yes
- [ ] No

## Risk Assessment
<!-- Assess the risk level of this PR:
- Low: Minimal impact, straightforward changes.
- Medium: Potential for some edge cases or indirect effects.
- High: Could affect critical functionality or many users.
-->
- [ ] Low
- [ ] Medium
- [ ] High

## Additional Notes
<!-- Add any other context or comments about the PR here -->

## Screenshots (if applicable)
<!-- Attach any screenshots that help explain your changes -->

martyu commented on app/src/main/java/com/flowfoundation/wallet/page/swap/dialog/select/SelectTokenDialog.kt:
```diff
@@ -208,12 +212,15 @@ class SelectTokenDialog : BottomSheetDialogFragment(), OnCoinRateUpdate {
         availableCoins: List<FlowCoin>? = null
     ) = suspendCoroutine { result ->
+        // Dismiss any existing instance of the dialog
+        (fragmentManager.findFragmentByTag(TAG) as? SelectTokenDialog)?.dismiss()
```
Won't this make it dismiss and reappear?  Couldn't you just return if the fragment is found?

jaymengxy replied:
You can directly record the time interval between clicks for the clicked view to control the issue of multiple clicks.

lealobanov commented:
https://github.com/user-attachments/assets/8cc4cb3f-cb18-498c-8910-de671cf668a6

Hi @jaymengxy @martyu , I've updated to add a time interval (500 ms) to avoid re-opening the dialog when clicked multiple times. Also added updates for https://github.com/onflow/FRW-Android/issues/738; `availableTokens` is passed straight away to the dialog, and updated prices are refreshed on screen when the values are received.

Thanks

lealobanov commented:
https://github.com/user-attachments/assets/e19420d6-8c51-469f-8821-f9b5de46499a
